src/Microsoft.SqlTools.ServiceLayer/QueryExecution/QueryExecutionService.cs

```diff
@@ -893,13 +894,18 @@ public async Task<ResultSetSubset> InterServiceResultSubset(SubsetParams subsetP
             // Setup the ResultSet updated callback
             ResultSet.ResultSetAsyncEventHandler resultUpdatedCallback = async r =>
             {
+                ExecutionPlanGraph graph = null;
+                if(r.Summary.Complete && r.Summary.SpecialAction.ExpectYukonXMLShowPlan == true)
+                {
+                    var xmlString = r.GetRow(0)[0].DisplayValue;
```

Comment:
I guess we always know that we can safely reference this element with `(0)[0]`.  i.e. no needs to bounds-check?